src/libs/Network/SequentialQueue.js

```diff
@@ -24,20 +40,42 @@ let currentRequest = null;
  * @returns {Promise}
  */
 function process() {
-    const persistedRequests = PersistedRequests.getAll();
+    const persistedRequests = [...PersistedRequests.getAll()];
```

Comment:
@madmax330 do you remember if there was a reason for that?

Reply:
+1

Reply:
Oh right I forgot about this. I'm going to try removing it and then I'll run the NetworkTests and manual tests to see if anything breaks.

Reply:
I imagine the reason for making a copy of the persisted requests is so that we don't modify the original request. We currently call shift on the array to remove the first value of the array, which means the request would be gone when we retry.

Instead of copying the whole array we can access the first value in the array and make a deep copy of that request object. Copying one request object is probably quicker than copying all of them.

I think we should pass a copy of the request object to the middlewares because they could modify the request and we probably want to prevent that given that it could be retried.  We should make a deep copy to ensure that the entire request is a copy, not just the first level. That way if someone modifies a deep property of the request in the middleware it won't actually get changed.

Maybe that's overkill though, hmm. I think it's best to make it hard to shoot ourselves in the foot, so I'm going to make a deep copy of the current request that's being processed.

Reply:
Yeah it's to avoid modifying the original array. Because `PersistedRequests.getAll()` just returns the variable, so when we modify it we'll be modifying the base array as well, which we don't want.